### DISCUSSION TITLE
Any way to access the current module's name or URI reflectively?

### DISCUSSION THREAD

**Discussion by raystubbs** [apple/pkl#336] [Q&A] [answered]
I'd like to set an implicit `id` field based on module name for each module that amends some parent.  Is there any good way to do this or something similar?

**Comment by bioball** [comment, marked as answer]
You can get this through the reflection API:

```
import "pkl:reflect"

local mod = reflect.Module(module)

moduleName = mod.name

moduleUri = mod.uri
```

More docs on the reflect API: https://pkl-lang.org/package-docs/pkl/current/reflect/index.html

**Reply by raystubbs** [reply]
Wow, awesome, thanks 🎉
